test_util/marathon.py

```diff
@@ -278,21 +278,15 @@ def deploy_pod(self, pod_definition, timeout=300):
                         retry_on_result=lambda ret: ret is False,
                         retry_on_exception=lambda x: False)
         def _wait_for_pod_deployment(pod_id):
-            # TODO(greggomann): Revisit this logic. Can we make a request for
-            # info on only our specific deployment? See DCOS-11707.
-            r = self.get('v2/deployments')
-            data = r.json()
-            if len(data) > 0:
-                log.info('Waiting for pod to be deployed %r', data)
-                return False
-            # deployment complete
-            r = self.get('v2/pods' + pod_id)
+            r = self.get('v2/pods' + pod_id + '::status')
             r.raise_for_status()
             data = r.json()
-            if int(data['scaling']['instances']) != pod_definition['scaling']['instances']:
-                log.info('Pod is still scaling. Continuing to wait...')
-                return False
-            return data
+            if 'status' in data and data['status'] == 'STABLE':
+                # deployment complete
+                return data
+            log.info('Waiting for pod to be deployed %r', data)
+            return False
```

Comment:
@adam-mesos, I synced with @jdef on this. In the context of the current helper function, the above behavior makes sense, since this is a simple one-shot pod deployment. When we have more complex pod tests, subscribing to the event bus will be more robust and we should switch to that in the long-term. I created a ticket here to track, and referenced it in a comment above: https://jira.mesosphere.com/browse/DCOS_OSS-1056